sklearn/cluster/_k_means_fast.pxd

```diff
@@ -0,0 +1,23 @@
+# cython: language_level=3
+
+
+from cython cimport floating
+cimport numpy as np
+
+
+cdef floating _euclidean_dense_dense(floating*, floating*, int, bint) nogil
+
+cdef floating _euclidean_sparse_dense(floating[::1], int[::1], floating[::1],
+                                      floating, bint) nogil
```

Comment:
These aren't tested (some others below as well). Maybe a few sanity checks wouldn't hurt?

Reply:
let's go

Reply:
I did it in the end :)